library/index.c

```diff
@@ -530,10 +530,14 @@ int eblob_generate_sorted_index(struct eblob_backend *b, struct eblob_base_ctl *
 	struct eblob_map_fd src, dst;
 	int fd, err, len;
 	char *file, *dst_file;
+	int cache_is_empty = 0;
 
 	if (b == NULL || bctl == NULL)
 		return -EINVAL;
 
+	/* check that eblob cache is empty. If cache is empty we can skip iterating and flushing it */
+	cache_is_empty = eblob_cache_empty(b);
+
```

Comment:
> Empty cache can be with unsorted index - sorting of the blob 1 has flushed the cache, sorting of the blob 2 starts without binary log, any update for that blob will corrupt data.

If we have situation when cache is empty and we have nonempty blob with unsorted index, no one can access/modify the data from the last blob, because we can't make a lookup in such blobs. If we have 2 unsorted blobs which keys are in the cache, after flushing blob 1 we still have blob 2's keys in the cache, because 'flushing  the cache' will remove only blob 1's keys from the cache.

> > It can't be made from these changes.
> Then where is the bug? I turned off automatic index sorting, there are no more suspicious commits.

The real bug should be carefully investigated but these changes can't cause it.

> I'm trying to figure out what caused the bug since 0.22.13 version, and you not only refuse to cooperate, you basically saying 'there is no bug'. Maybe you are right, but I can not take it for granted, thus I change code and will do this until bug has been found.
> 
> The benefit is a getting rid of useless iteration thru all blob's keys. It depends on number of keys.
No numbers? Index iteration is a matter of seconds, but I agree it has its cost even at startup time, so I propose a change which will check cache size at flushing time under its current lock and if cache is empty, flush will just return.

Everything is a matter of seconds, even unixtime is number of seconds from 1 January 1970. As I already said, it linearly depends on a number of keys in blob. And this dependence can dramatically increase time of eblob initialization.